feature/main/src/main/java/com/ku_stacks/ku_ring/main/search/compose/SearchScreen.kt

```diff
@@ -81,8 +83,8 @@ private fun SearchScreen(
     onClickSearch: (SearchState) -> Unit,
     searchState: SearchState,
     tabPages: List<SearchTabInfo>,
-    noticeSearchResult: SharedFlow<List<Notice>>,
-    staffSearchResult: SharedFlow<List<Staff>>,
+    noticeSearchResult: StateFlow<List<Notice>>,
```

Comment:
하위 Composable에서 Flow에 직접 접근해야 할 특별한 이유가 없다면, `Flow.collectAsState()` API를 이용하여 Flow 내부의 값을(여기서는 `List<Notice>`) 전달하는 게 어떨까요? (값 대신 Flow를 전달해야 할 이유가 있는가?)

Reply:
의도는 noticeSearchResult 를 최상단에 val 로서 들고있으면 검색해서 변경될 때마다 전체 recomposition 이 발생하지않을까해서였어요. 이렇게 SharedFlow 로서 말단까지 전달했을 때 noticeList 의 변경이 생기면 NoticeSearchScreen 만 recomposition 이 발생하여 이점이 있지않을까용?

Reply:
재할당된 매개변수를 사용하는 composable만 recomposition되므로, StateFlow를 넘겨주든 StateFlow.value를 넘겨주든 recomposition 범위는 동일합니다.

여기까지가 `noticeSearchResult`의 매개변수 타입에 관련된 내용입니다.

---

여기서부터는 일반적인 recomposition 관련 내용입니다.

사실 `noticeSearchResult`의 타입을 무엇으로 하더라도, `NoticeSearchScreen`만 recomposition되는 건 아닙니다. 왜냐하면 `List`를 포함한 composable은 항상 recomposition되기 때문입니다. (`SearchResultHorizontalPager` 등)

그 이유는 Compose 컴파일러가 `List`, `Set` 등 일반적인 collections을 unstable하다고 판단하기 때문입니다. 공식 문서에서 자세히 설명하고 있으니, 읽어보시면 좋을 것 같아요

https://developer.android.com/jetpack/compose/performance/stability/fix#immutable-collections

> The Compose compiler has marked Snack as unstable. This is because the type of the tags parameter is Set<String>. This is an immutable type, given that it is not a MutableSet. However, standard collection classes such as Set, List, and Map are ultimately interfaces. As such, the underlying implementation may still be mutable.

> For example, you could write val set: Set<String> = mutableSetOf("foo"). The variable is constant and its declared type is not mutable, but its implementation is still mutable. The Compose compiler cannot be sure of the immutability of this class as it only sees the declared type. It therefore marks tags as unstable.

Reply:
구두로 얘기한것처럼 상위 스크린에서 recomposition 이 발생하지만 skip 되어 성능적인 차이가 없으니 바로 List로 적용하였습니다. 
[197320f](https://github.com/ku-ring/KU-Ring-Android/pull/96/commits/197320f1ea19e7e96a571a924b971a4e9d8405e1) 에 반영했습니다~
의견감사합니다~!